### ISSUE TITLE
Floatlit fails with 27f32

### ISSUE THREAD

**Issue by eugenesvk** [LukasKalbertodt/litrs#14]
The docs on floats https://docs.rs/litrs/latest/litrs/struct.FloatLit.html say that `27f32` is a float

> A floating point literal, e.g. 3.14, 8., 135e12, 27f32 or 1.956e2f64.

Yet this fails with `failed to parse as float literal: ParseError { span: None, kind: UnexpectedIntegerLit }`
```rs
use litrs::{FloatLit};
let float_lit = FloatLit::parse("27f32").expect("failed to parse as float literal");
```
`27.0f32` succeeds

**Comment by dtolnay** [comment]
According to the Rust Reference, `27f32` is a "decimal integer" literal (with a suffix).

https://doc.rust-lang.org/1.73.0/reference/tokens.html#numbers

**Comment by eugenesvk** [comment]
To be more precise, it's an "integer literal, accepted as a floating-point literal expression",  https://doc.rust-lang.org/1.73.0/reference/tokens.html#number-literals and given that it's a "floating-point literal expression" I don't see why the floating point parser can't parse it to a float even it it's also possible to parse it to an integer

**Comment by LukasKalbertodt** [comment]
This is indeed a bugs in the docs. In 0.4.0 I fixed this inconsistency with the spec and implemented it as specified: [0.4.0 changelog](https://github.com/LukasKalbertodt/litrs/blob/main/CHANGELOG.md#changed). Will fix the docs soon.

I considered (and am still open to) adding `FloatLit::parse_lenient` or something like that, which would accept integer literals as well. It's not possible to make `Literal::parse` return `FloatLit` for `27f32`, that would just be wrong according to the spec. One could argue `FloatLit::parse` could just accept `27f32`, but I wanted to keep it consistent with `Literal::parse`. If you're interested in `parse_lenient` I will add it.

And to clarify something regarding "floating-point literal *expression*": there are multiple places where the Rust compiler does work after the tokenization phase. `-128i8` for example is lexed as `-` (punct) and `128i8` (literal). Of course, 128 is not representable as `i8`, but of course you can write that in Rust. Designing `litrs`, it's a bit difficult to chose between convenience and "lexer spec" complience here. Not accepting negative literals for example (which don't exist in the Rust lexocographical grammar!) is probably annoying. But I think adding more `_lenient` functions is a better option than making the default code path different from the spec.

**Comment by eugenesvk** [comment]
> But I think adding more `_lenient` functions is a better option than making the default code path different from the spec

Those would be quite nice, especially if they would also allow the minus sign which I currently have to parse manually (that's indeed annoying)! 

(don't know what option is best, but either way would work)
